Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm

```diff
@@ -3885,41 +3885,41 @@ static NSStringCompareOptions compareOptionsForFindOptions(WebCore::FindOptions
 
 void UnifiedPDFPlugin::clearSelection()
 {
+    resetInitialSelection();
     setCurrentSelection({ });
 }
 
+static bool areVisuallyDistinct(FloatPoint a, FloatPoint b)
+{
+    static constexpr auto maxDistanceSquared = 0.1 * 0.1;
+    return (a - b).diagonalLengthSquared() > maxDistanceSquared;
+}
+
 void UnifiedPDFPlugin::setSelectionRange(FloatPoint pointInRootView, TextGranularity granularity)
 {
 #if HAVE(PDFDOCUMENT_SELECTION_WITH_GRANULARITY)
     RetainPtr pdfDocument = m_pdfDocument;
     if (!pdfDocument)
         return;
 
-    auto pdfGranularity = [granularity] {
-        switch (granularity) {
-        case TextGranularity::CharacterGranularity:
-            return PDFSelectionGranularityCharacter;
-        case TextGranularity::WordGranularity:
-            return PDFSelectionGranularityWord;
-        case TextGranularity::LineGranularity:
-            return PDFSelectionGranularityLine;
-        default:
-            ASSERT_NOT_REACHED();
-            return PDFSelectionGranularityCharacter;
-        }
-    }();
-
     auto [page, pointInPage] = rootViewToPage(pointInRootView);
     if (!page)
         return;
 
-    setCurrentSelection([pdfDocument selectionFromPage:page.get() atPoint:pointInPage toPage:page.get() atPoint:pointInPage withGranularity:pdfGranularity]);
+    m_initialSelection = selectionAtPoint(pointInPage, page.get(), granularity);
+    m_initialSelectionStart = selectionCaretPointInPage(m_initialSelection.get(), SelectionEndpoint::Start);
+    setCurrentSelection(m_initialSelection.get());
 #else
     UNUSED_PARAM(pointInRootView);
     UNUSED_PARAM(granularity);
 #endif
 }
 
+static bool isEmpty(PDFSelection *selection)
+{
+    return !selection.pages.count;
+}
```

Comment:
There's a `-[PDFSelection isEmpty:]` method, albeit SPI.

Reply:
We should probably just use that instead?

Reply:
When I originally used `pages.count` we didn't have the `isEmpty:` method available.

Reply:
Ah, yeah — that's why I opted for this approach which only uses public API (instead of `-isEmpty:` which uses SPI).

Reply:
Makes sense